src/devices/src/virtio/block/request.rs

```diff
@@ -227,13 +227,11 @@ impl Request {
                 METRICS.block.write_bytes.add(self.data_len as usize);
                 METRICS.block.write_count.inc();
             }
-            RequestType::Flush => match diskfile.flush() {
-                Ok(_) => {
-                    METRICS.block.flush_count.inc();
-                    return Ok(0);
-                }
-                Err(e) => return Err(ExecuteError::Flush(e)),
-            },
+            RequestType::Flush => {
+                diskfile.sync_all().map_err(ExecuteError::SyncAll)?;
+                METRICS.block.flush_count.inc();
+                return Ok(0);
```

Comment:
>Is this needed?
What exactly are you referring to? The code itself, or the question about the integration test? 

I did not do any additional tests yet, as they "should" have already been covered by the existing test framework. I will  look at adding a specific test somehow for data loss between VM and host when the virtio flush fails or is non-existent.

Reply:
Oh, no, I was referring to that return statement. We return `Ok(0)` anyway at the end.

Reply:
Not sure if [this](https://www.thomas-krenn.com/en/wiki/Linux_Page_Cache_Basics) - see *Functional Approach/Writing* section - helps, but here is an example of how we can force flushing to disk. Not sure though if this can be used as a regression, though.

Reply:
Sorry, I did not understand. Yes, you are correct that is not needed. I will remove it.

Reply:
With regard to a functional test, it will be a bit tricky. What has to happen is that without the sync, we have to read the data on the host from the device itself, bypassing the page cache and verify that it is different than what was written in the guest.  I don't know if `O_DIRECT` on an ext4 filesystem will cause dirty pages to be flushed before accessing, but that might be one way. 

I think `madvise(MADV_REMOVE)` will flush dirty pages before punching a hole, so that most likely will not work. `madvise(MADV_FREE)` might work if it really drops dirty pages without writing; then the data could be refreshed from the physical media and compared to some known pattern.